### PR TITLE
docs(start_planner): add start/end condition

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

update README for start_planner
This PR mainly includes description of start/end condition of module

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/README.md [hunk 1]
```diff
@@ -73,6 +73,207 @@ Here are some notable limitations:
...
+
+### When approval is required?
+
+#### Forward driving
+
+- **Start approval required**: Even if a route is generated, approval is required to initiate movement. If a dynamic object poses a risk, such as an approaching vehicle from behind, candidate routes may be displayed, but approval is necessary for departure.
+
+#### Backward driving + forward driving
+
+- **Multiple approvals required**: When the planned route includes a backward driving, multiple approvals are needed before starting the reverse and again before restarting forward movement. Before initiating forward movement, the system conducts safety checks against dynamic obstacles. If a detection is detected, approval for departure is necessary.
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/README.md:273]: path is better than route.
and same to other parts

File: planning/behavior_path_start_planner_module/README.md [hunk 2]
```diff
@@ -73,6 +73,207 @@ Here are some notable limitations:
...
+
+  - Why is a margin from the rear object necessary?
+    For objects ahead, another behavior module can intervene, allowing the path to overwrite itself through an avoidance plan, even if the clearance from the path to a static obstacle is minimal, thus maintaining a safe distance from static obstacles. However, for objects behind the vehicle, it is impossible for other behavior modules other than the start_planner to alter the path to secure a margin, potentially leading to a deadlock by an action module like "obstacle_stop_cruise" and subsequent immobilization. Therefore, a margin is set for stationary objects at the rear.
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/README.md:143]: obstacle_stop_cruise -> obstacle_cruise_planner ?


obstacle_stop_planner can not stop along backward path
https://github.com/autowarefoundation/autoware.universe/issues/1697

File: planning/behavior_path_start_planner_module/README.md (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/README.md]: changed in https://github.com/autowarefoundation/autoware.universe/pull/6329/commits/3b87e239c805ddb0c6e2f29eddff48a20f1aa04e

File: planning/behavior_path_start_planner_module/README.md (hunk 2 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/README.md]: changed in https://github.com/autowarefoundation/autoware.universe/pull/6329/commits/1d52778a7520c7986122850bdeee3161befaafbf

**Review by kosuke55 — APPROVED** [review]
LGTM thanks!

**Pull request merged**